test-suite/src/test/scala/org/scalajs/testsuite/javalib/ConcurrentLinkedQueueTest.scala

```diff
@@ -0,0 +1,148 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package  org.scalajs.testsuite.javalib
+
+import scala.language.implicitConversions
+
+import scala.collection.JavaConversions._
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.runtime.UndefinedBehaviorError
+
+import java.util.concurrent.ConcurrentLinkedQueue
+
+object ConcurrentLinkedQueueTest extends JasmineTest {
+
```

Comment:
You should extend the `AbstractCollectionTest` class or even better if you create an `AbstractQueueTest` that extends `AbstractCollectionTest` and extend this test to that one. 

If you do that you may need to define a `def allowsNullElements: Boolean` in `CollectionFactory` and special case some of the tests.